src/index.js

```diff
@@ -98,16 +104,26 @@ class FloodSub extends BaseProtocol {
     }
   }
 
-  _processRpcMessages (msgs) {
-    msgs.forEach((msg) => {
-      const seqno = utils.msgId(msg.from, msg.seqno)
-      // 1. check if I've seen the message, if yes, ignore
-      if (this.seenCache.has(seqno)) {
+  /**
+   * @private
+   * @param {rpc.RPC.Message} message The message to process
+   * @returns {void}
+   */
+  _processRpcMessage (message) {
+    const msg = utils.normalizeInRpcMessage(message)
+    const seqno = utils.msgId(msg.from, msg.seqno)
+    // 1. check if I've seen the message, if yes, ignore
+    if (this.seenCache.has(seqno)) {
+      return
+    }
+
+    this.seenCache.put(seqno)
+    this.validate(message, (err, isValid) => {
```

Comment:
Could you add an extra step in the comments? Otherwise, the remaining comments don't seem helpful.

Reply:
done!